etcdserver/api/membership/cluster.go

```diff
@@ -247,7 +247,12 @@ func (c *RaftCluster) Recover(onSet func(*zap.Logger, *semver.Version)) {
 	defer c.Unlock()
 
 	c.members, c.removed = membersFromStore(c.lg, c.v2store)
-	c.version = clusterVersionFromStore(c.lg, c.v2store)
+	if c.be != nil {
+		c.version = clusterVersionFromBackend(c.lg, c.be)
+	} else {
+		c.version = clusterVersionFromStore(c.lg, c.v2store)
```

Comment:
Do we need fallback on v2 version data if v3 version is `nil` (upgraded etcd process)?

```go
c.version = clusterVersionFromStore(c.lg, c.v2store)
if c.be != nil {
        v3Ver := clusterVersionFromBackend(c.lg, c.be)
	if v3Ver != nil { c.version = v3Ver }
}
```

Reply:
Not sure about it. When will v3 version data is nil while v2 is not? If v3 backend is available, the v3 version will not be nil except starting a new server? Setting version will store new version info into both v2 store and v3 backend.
https://github.com/etcd-io/etcd/blob/1f8764be3b43448ccfd60706c42dab09b0bc6ed3/etcdserver/api/membership/cluster.go#L571-L576

Reply:
Ok, that answers my question. Thanks.